### PR TITLE
Consensus metrics error

### PR DESCRIPTION
**Pull request by koseoyoung**
works on #1501 
not return the error, but print out log when it is an empty blockchain

### CODE REVIEW

**Comment by zjshen14** [comment]
To fix the problem, is it as simple as that when height = 0, we default the candidate list to empty?

File: server/itx/heartbeat.go [hunk 1]
```diff
@@ -111,6 +114,10 @@ func (h *HeartbeatHandler) Log() {
 			// RollDpos Concensus Metrics
 			consensusMetrics, err = rolldpos.Metrics()
 			if err != nil {
+				if height == 0 && errors.Cause(err) == statedb.ErrStateNotExist {
+					log.L().Warn("consensus metrics are not ready yet, candidates are empty")
+					return
```
Comment (CoderZhi) [inline server/itx/heartbeat.go]: is it intended to return error in this case?

Comment (koseoyoung) [reply server/itx/heartbeat.go]: Oh no, I think we don't have to return in here

File: server/itx/heartbeat.go [hunk 2]
```diff
@@ -111,8 +114,12 @@ func (h *HeartbeatHandler) Log() {
 			// RollDpos Concensus Metrics
 			consensusMetrics, err = rolldpos.Metrics()
 			if err != nil {
-				log.L().Error("failed to read consensus metrics", zap.Error(err))
-				return
+				if height == 0 && errors.Cause(err) == statedb.ErrStateNotExist {
```
Comment (zjshen14) [inline server/itx/heartbeat.go]:
```
if height > 0 || errors.Cause(err) != statedb.ErrStateNotExist {
	log.L().Error("failed to read consensus metrics", zap.Error(err))
	return
}
```

**Pull request merged**